syscalls_darwin.go

```diff
@@ -159,7 +159,16 @@ func (t *tunReadCloser) Write(from []byte) (int, error) {
 	}
 	t.wBuf = t.wBuf[:len(from)+4]
 
-	t.wBuf[3] = 2 // Family: IP (2)
+	// Determine the IP Family for the NULL L2 Header
+	ipVer := from[0] >> 4
```

Comment:
Good point! Since we are simulating the reverse of kernel's packet header logic here, I guess we'd need to read kernel code to figure out what the expected behavior is when an empty buffer is written to a tun FD created without `IFF_NO_PI` flag.